src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/ParserHelpers.cs

```diff
@@ -43,12 +38,32 @@ public static bool IsIdentifier(string value)
 
     public static bool IsIdentifier(string value, bool requireIdentifierStart)
     {
-        IEnumerable<char> identifierPart = value;
+        var span = value.AsSpan();
+
         if (requireIdentifierStart)
         {
-            identifierPart = identifierPart.Skip(1);
+            if (!IsIdentifierStart(span[0]))
```

Comment:
Do we need to guard against a zero length span here?

Reply:
For sure. It didn't before, but that doesn't mean we shouldn't.